kernel-ewasm/cap9-std/src/proc_table/cap.rs

```diff
@@ -98,6 +109,201 @@ impl Capability {
             Capability::AccountCall(_) => CAP_ACC_CALL_SIZE,
         }
     }
+
+    pub fn cap_type(&self) -> u8 {
+        match self {
+            Capability::ProcedureCall(_) => CAP_PROC_CALL,
+            Capability::ProcedureRegister(_) => CAP_PROC_REGISTER,
+            Capability::ProcedureDelete(_) => CAP_PROC_DELETE,
+            Capability::ProcedureEntry(_) => CAP_PROC_ENTRY,
+            Capability::StoreWrite(_) => CAP_STORE_WRITE,
+            Capability::Log(_) => CAP_LOG,
+            Capability::AccountCall(_) => CAP_ACC_CALL,
+        }
+    }
+
+    pub fn is_subset_of(&self, parent_cap: &Capability) -> bool {
```

Comment:
Looks like `get_proc_cap` without generics is adding additional code-smell here.

Reply:
Likely a trait would work here similar to the [Byteorder](https://docs.rs/byteorder/1.3.2/byteorder/) Trait pattern.

Reply:
Actually scratch the above, I misunderstood.